### PR TITLE
Update Nemo launcher to GA version

### PR DESCRIPTION
**Pull request by verdimrc**
*Issue #, if available:* N/A

*Description of changes:* nemo-launcher: update to GA image.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 1]
```diff
@@ -47,21 +71,40 @@ RUN mkdir -p /tmp && \
...
+ENV FI_PROVIDER=efa
+
+# p5 must not set NCCL_PROTO=simple.
+# See https://github.com/aws/aws-ofi-nccl/commit/3d5e21db5f2b0f307c5d0c206c4b1b66d72d0d04
+# ENV NCCL_PROTO=simple
 
 ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
+
+# TODO: validate this, based on Based on https://github.com/NVIDIA/NeMo-Megatron-Launcher/blob/7e482f60e00f266a75b9a6964da2afdd2ebe665d/csp_tools/aws/Dockerfile
+#ENV FI_PROVIDER=efa \
```
Comment (perifaws) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:98]: remove as well?

**Review by perifaws — COMMENTED** [review]
A few commented things to remove

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 2]
```diff
@@ -6,36 +6,60 @@
 
 FROM nvcr.io/ea-bignlp/nemofw-training:23.07-py3
 
-ARG DEBIAN_FRONTEND=noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
 ENV EFA_INSTALLER_VERSION=latest
-ENV NCCL_VERSION=inc_nsteps
-ENV AWS_OFI_NCCL_VERSION=1.4.0-aws
+#ENV NCCL_VERSION=inc_nsteps
```
Comment (perifaws) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:11]: Remove if commented?

**Review by perifaws — CHANGES_REQUESTED** [review]

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 3]
```diff
@@ -6,36 +6,60 @@
 
 FROM nvcr.io/ea-bignlp/nemofw-training:23.07-py3
 
-ARG DEBIAN_FRONTEND=noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
```
Comment (perifaws) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:9]: good catch

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 4]
```diff
@@ -6,36 +6,60 @@
...
+    curl \
+    autoconf \
+    libtool \
+    gdb \
+    automake \
+    cmake \
+    apt-utils \
+    libhwloc-dev \
+    aptitude && \
+    DEBIAN_FRONTEND=noninteractive apt autoremove -y
```
Comment (perifaws) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:40]: shall we keep `DEBIAN_FRONTEND=noninteractive` on the `apt-get` commands? Seems redundant with the `ENV` line 9.

**Review by perifaws — CHANGES_REQUESTED** [review]

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile (hunk 1 above)
Comment (verdimrc) [reply 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile]: to validate which paths still need to set minimally.

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 5]
```diff
@@ -47,21 +72,33 @@ RUN mkdir -p /tmp && \
...
+    OMPI_MCA_btl_tcp_if_exclude=lo,docker0 \
+    OPAL_PREFIX=/opt/amazon/openmpi \
+    NCCL_SOCKET_IFNAME=^docker,lo   \
+    FI_EFA_USE_HUGE_PAGE=0
 
-ENV RDMAV_FORK_SAFE=1             \
-    FI_PROVIDER=efa               \
-    FI_EFA_USE_DEVICE_RDMA=1      \
-    NCCL_PROTO=simple
+# TODO: validate this, based on https://github.com/NVIDIA/NeMo-Megatron-Launcher/blob/7e482f60e00f266a75b9a6964da2afdd2ebe665d/csp_tools/aws/Dockerfile
```
Comment (sean-smith) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:92]: remove please

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile [hunk 6]
```diff
@@ -47,21 +72,33 @@ RUN mkdir -p /tmp && \
...
+#ENV LD_LIBRARY_PATH="/usr/local/lib:/opt/nccl/build/lib:/usr/local/cuda/lib64:/opt/amazon/efa/lib:/opt/amazon/openmpi/lib:/opt/aws-ofi-nccl/lib:${LD_LIBRARY_PATH}" \
+#    PATH="/usr/local/nvm/versions/node/v16.19.1/bin:/opt/amazon/efa/bin:/usr/local/lib/python3.8/dist-packages/torch_tensorrt/bin:/opt/amazon/openmpi/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/ucx/bin:/opt/tensorrt/bin:/opt/s5cmd"
 
-ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
+# NCCL-tests
+RUN git clone https://github.com/NVIDIA/nccl-tests.git /opt/nccl-tests \
```
Comment (sean-smith) [inline 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:97]: Do we need the nccl tests?

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile (hunk 5 above)
Comment (verdimrc) [reply 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:92]: ack, it'll be removed, or minimized, after testing to confirm which additional path + ldpath still needed.

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile (hunk 6 above)
Comment (verdimrc) [reply 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile:97]: to me, it's a best practice to include nccl-test as the first-line diagnostic tool to validate container correctness.

**Comment by perifaws** [comment]
Is the PR still current?

**Comment by verdimrc** [comment]
yes, it's based on the GA version of nemo container.

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile (hunk 1 above)
Comment (verdimrc) [reply 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile]: yep, not needed. Base image already takes care of this.

File: 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile (hunk 5 above)
Comment (verdimrc) [reply 3.test_cases/2.nemo-launcher/0.NemoMegatron-aws-optimized.Dockerfile]: Removed, parent image already takes care of these.

**Comment by verdimrc** [comment]
@perifaws, @sean-smith: ready for review. All changes tested to work.

**Comment by perifaws** [comment]
Let's squash the commits, it's only a few changes and 11 commits. Good to go otherwise.

**Review by perifaws — APPROVED** [review]

**Pull request merged**